src/functional/ffront/decorator.py

```diff
@@ -480,6 +482,7 @@ def from_function(
             **operator_attribute_nodes,
         )
         foast_node = FieldOperatorTypeDeduction.apply(untyped_foast_node)
+        assert isinstance(foast_node, (foast.ScanOperator, foast.FieldOperator))
```

Comment:
related to the comment below this should be an instance of `OperatorNodeT`, not sure how mypy deals with that.